larpandoracontent/LArControlFlow/StitchingCosmicRayMergingTool.cc

```diff
@@ -538,7 +556,27 @@ void StitchingCosmicRayMergingTool::StitchPfos(const MasterAlgorithm *const pAlg
             }
 
             for (const ParticleFlowObject *const pPfoToShift : pfoVector)
+            {
+                CaloHitList caloHitList3D;
+                LArPfoHelper::GetCaloHits(pPfoToShift, TPC_3D, caloHitList3D);
+
+                float xsum(0.f);
+                int count(0);
+
+                for (const CaloHit *const pCaloHit : caloHitList3D)
+                {
+                    xsum += pCaloHit->GetPositionVector().GetX();
+                    ++count;
+                }
+
+                if (count == 0)
+                    continue;
+
+                const float averageX(xsum / static_cast<float>(count));
```

Comment:
I would personally remove the variable `count` (just another thing that could go wrong), get `xSum` (should ideally have capital S!) in a for loop with a single line body, i.e. without parentheses and check `caloHitList3D.size()` (could do this check first, and cache value in a const unsigned int, and use it in the calculation of `averageX`).

I wondered whether this was just another way of characterising a cluster position (we probably have too many different ways to do this), but mean X is just so simple, we probably don't need to worry here.

Reply:
Good idea, caching the int first and then checking the size is more efficient.  The xsum has now been changed to xSum, which I'm very happy to do, but I think I copied this from a helper (I think) so we might need to update that too in the future!